lib/checknullpointer.cpp

```diff
@@ -509,31 +509,49 @@ void CheckNullPointer::arithmetic()
     for (std::size_t i = 0; i < functions; ++i) {
         const Scope * scope = symbolDatabase->functionScopes[i];
         for (const Token* tok = scope->classStart->next(); tok != scope->classEnd; tok = tok->next()) {
-            if (!tok->astOperand2() || tok->str() != "-")
-                continue;
-            // pointer subtraction
-            if (!tok->valueType() || !tok->valueType()->pointer)
-                continue;
-            // Can LHS be NULL?
-            const ValueFlow::Value *value = tok->astOperand1()->getValue(0);
-            if (!value)
-                continue;
-            if (!_settings->inconclusive && value->isInconclusive())
-                continue;
-            if (value->condition && !_settings->isEnabled(Settings::WARNING))
-                continue;
-            arithmeticError(tok,value);
+            if (tok->str() == "-" || tok->str() == "+" || tok->str() == "+=" || tok->str() == "-=")
+            {
+                const Token *arguments[] = {tok->astOperand1(), tok->astOperand2()};
+                for(const Token *argument:arguments) {
```

Comment:
Unfortunately, we have not started using range for loops yet. We want to support some old compilers that only has partial c++11 support.

Reply:
What C++11 features are allowed?

Reply:
c++11 should be mostly avoided. off the top of my head we do use; nullptr, long long, r-value references, right angle brackets. The nullptr can be emulated easily if the compiler does not support it.

Even though I'd say we have been very conservative, people are sometimes contacting me needing help to compile with old versions of GCC.